### ISSUE TITLE
Cost nan

### ISSUE THREAD

**Issue by TexasRangers86** [PaddlePaddle/Paddle#5012]
The code runs all right, but when I change the input data, I got a nan problem!
`import sys
import paddle.v2 as paddle
import cPickle
import copy
import os
import numpy as np

with_gpu = os.getenv('WITH_GPU', '0') != '0'

def get_usr_combined_features():
    user_base = paddle.layer.data(
        name='user_base',
        type=paddle.data_type.dense_vector(10))

    user_topic = paddle.layer.data(
        name='user_topic', type=paddle.data_type.dense_vector(2589))
    usr_topic_fc = paddle.layer.fc(input=user_topic, size=512)

    usr_combined_features = paddle.layer.fc(
        input=[user_base, usr_topic_fc],
        size=128,
        act=paddle.activation.Tanh()
        # act = paddle.activation.Relu()
        )
    return usr_combined_features

def get_content_combined_features():
    content_ctr = paddle.layer.data(
        name='content_ctr',
        type=paddle.data_type.dense_vector(1))

    content_product = paddle.layer.data(
        name='content_product',
        type=paddle.data_type.dense_vector(1))

    content_plsa = paddle.layer.data(
        name='content_plsa',
        type=paddle.data_type.dense_vector(1000))
    content_plsa_fc = paddle.layer.fc(content_plsa, size=256)

    content_combined_features = paddle.layer.fc(
        input=[content_ctr, content_product, content_plsa_fc],
        size=128,
        act=paddle.activation.Tanh()
        # act=paddle.activation.Relu()
        )
    return content_combined_features

def main():
    #paddle.init(use_gpu=with_gpu)
    paddle.init(use_gpu=True, trainer_count=1)
    #paddle.init(use_gpu=False, trainer_count=2)
    usr_combined_features = get_usr_combined_features()
    content_combined_features = get_content_combined_features()
    inference = paddle.layer.cos_sim(
        a=usr_combined_features, b=content_combined_features, size=1)
    cost = paddle.layer.square_error_cost(
        input=inference,
        label=paddle.layer.data(
            name='score', type=paddle.data_type.dense_vector(1)))

    parameters = paddle.parameters.create(cost)
    trainer = paddle.trainer.SGD(
        cost=cost,
        parameters=parameters,
        update_equation=paddle.optimizer.Adam(learning_rate=1e-4))

    feeding = {
        'user_base': 0,
        'user_topic': 1,
        'content_ctr': 2,
        'content_product': 3,
        'content_plsa': 4,
        'score': 5
    }
        if isinstance(event, paddle.event.EndPass):
            print "Test %d, Cost %f" % (event.pass_id, result.cost)
            with gzip.open(path, 'w') as f:
                parameters.to_tar(f)

    def norm(data, dim, scale):
        return [float(data[i])/scale for i in range(dim)]
        with open("./paddle_erised_plsa.train") as file:
            for line in file:
                data = line.strip().split("\t")
                yield map(int, data[37:47]), norm(data[2074:4663], 2589, 100), [float(data[1])], [int(data[2])], norm(data[1047:2047], 1000, 1), [int(data[0])]

    def erised_plsa_read_test():
        with open("./paddle_erised_plsa.test") as file:
            for line in file:
                data = line.strip().split("\t")
                yield map(int, data[37:47]), norm(data[2074:4663], 2589, 100), [float(data[1])], [int(data[2])], norm(data[1047:2047], 1000, 1), [int(data[0])]

    trainer.train(
        reader=paddle.batch(
            reader=paddle.reader.shuffle(erised_plsa_read_train, 8192),
            batch_size=256),
        event_handler=event_handler,
        feeding=feeding,
        num_passes=200)

if __name__ == '__main__':
    main()`

When I set the batch_size to 20 or less, it runs all right.

**Comment by NHZlX** [comment]
try to change 
```
trainer = paddle.trainer.SGD(
    cost=cost,
    parameters=parameters,
    update_equation=paddle.optimizer.Adam(learning_rate=1e-4))
```
to 

```
trainer = paddle.trainer.SGD(
    cost=cost,
    parameters=parameters,
    update_equation=paddle.optimizer.Adam(learning_rate=1e-4/batch_size))
```

**Comment by TexasRangers86** [comment]
It seems unuse, and I try to set `reader=paddle.reader.shuffle(erised_plsa_read_train, 8192)` the buf size to 256, like this `reader=paddle.reader.shuffle(erised_plsa_read_train, 256)`, it works well.

**Comment by TexasRangers86** [comment]
hi I have solved this problem,  the reason is some layer input defined as dense_vector got all {0} vector. I filter these samples, and it works well now;

**Comment by 357589873** [comment]
hi,i want to know how to find layer input got {0} vector?
